share/functions/__fish_toggle_comment_commandline.fish

```diff
@@ -11,6 +11,7 @@ function __fish_toggle_comment_commandline --description 'Comment/uncomment the
     set -l cmdlines (commandline -b)
     if test "$cmdlines" = ""
         history search -p "#" -z | read -z cmdlines
+        set cmdlines (string split -- \n $cmdlines)
```

Comment:
`set cmdlines (history search -p "#" --max=1)` is simpler

Reply:
Awesome. To be honest I'm not familiar with `history`'s options and I was all thinking about "how to fix the incorrect command" instead of "correctly getting the command".